spec/controllers/restaurants_controller_spec.rb

```diff
@@ -0,0 +1,106 @@
+require_relative '../../app/controllers/restaurants_controller'
+require 'json'
+
+
+describe RestaurantsController, type: :controller do
+  describe "GET index" do
+    it "has a 200 status code" do
+      get :index
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe '#create' do
+    it "creates a new restaurant" do
+      post :create, params: { name: 'Artzieli11',
+                                    cuisine: 'pizza',
+                                    rating: 3,
+                                    ten_bis: true,
+                                    address: 'street in tel aviv',
+                                    max_delivery_time: 45 }
+      expect(response.status).to eq(200)
+    end
+
+    it "can't create a restaurant with an invalid rating" do
+      post :create, params: { name: 'Artzieli',
+                              cuisine: 'pizza',
+                              rating: 7,
+                              ten_bis: true,
+                              address: 'street in tel aviv',
+                              max_delivery_time: 45 }
+      expect(response.status).to eq(400)
+    end
+
+    it "creates a restaurant without optional fields" do
+      post :create, params: { name: 'Artzieli', address: 'street in tel aviv' }
+      expect(response.status).to eq(200)
+    end
+
+    it "can't create a restaurant without a name" do
+      post :create, params: { address: 'street in tel aviv' }
+      expect(response.status).to eq(400)
+    end
+
+    it "can't create a restaurant with a name that already exists" do
+      post :create, params: { name: 'Artzieli1',
```

Comment:
Another way could be letting FactoryBot doing all params work for you. Something like 
```
let (:rest) {FactoryBot.build(:restaurant)} # notice I used *build* which does not persist this instance
      post :create, params: { rest.to_params } # not sure this is the syntax for object->params, but you get the idea...
```